Update docs for plugin submission form (#592)
grafana/plugin-tools: docusaurus/docs/publish-a-plugin/publish-or-update-a-plugin.md

1. In the left menu, under Org Settings, click **My Plugins**.
1. Click **Submit New Plugin**. The Create Plugin Submission dialog box appears.

>    ![Submit plugin.](/img/plugins-submission-create.png)

Reviewer: The image returns a 404. The correct URL is:

```suggestion
   ![Submit plugin.](/img/plugin-submission-create.png)
```

Also the old image (`img/plugins-submission-create2.png`) is not used anymore and should be deleted

Reviewer 2: think this is fixed now?

Reviewer: Yep the link has been fixed :raised_hands: 

However the old image `img/plugins-submission-create2.png` should be deleted.

   - **URL:** A URL that points to a ZIP archive of your packaged plugin.
   - **Source code URL:** A URL that points to a public Git repository or ZIP archive of your complete plugin source code.
   - **SHA1:** The SHA1 hash of the plugin specified by the **URL**.
   - **Testing guidance:** An overview covering the installation, configuration, and usage of your plugin.

>    - **Provisioning provided for test environment:** It indicates that you've added provisioning functionality to your plugin.

Reviewer 2: ```suggestion
   - **Provisioning provided for test environment:** Check this box if you have [configured provisioning](./provide-test-environment.md).
```
